### PR TITLE
Remove RAD

### PR DESCRIPTION
**Pull request by CodeSandwich**
Remove RAD and add burning to Radicle Token

Part of #95

### CODE REVIEW

File: contracts/Governance/RadicleToken.sol [hunk 1]
```diff
@@ -187,6 +187,35 @@ contract RadicleToken {
...
+                    "RadicleToken::burnFrom: burned amount exceeds spender allowance"
+                );
+            allowances[account][spender] = newAllowance;
+            emit Approval(account, spender, newAllowance);
+        }
+
+        balances[account] = sub96(
+            balances[account],
+            amount,
+            "RadicleToken::_burnTokens: burned amount exceeds balance"
```
Comment (cloudhead) [inline contracts/Governance/RadicleToken.sol:210]:
```suggestion
            "RadicleToken::burnFrom: burn amount exceeds balance"
```

File: contracts/Governance/RadicleToken.sol [hunk 2]
```diff
@@ -187,6 +187,35 @@ contract RadicleToken {
...
+        uint96 amount = safe96(rawAmount, "RadicleToken::approve: amount exceeds 96 bits");
+
+        address spender = msg.sender;
+        uint96 spenderAllowance = allowances[account][spender];
+        if (spender != account && spenderAllowance != uint96(-1)) {
+            uint96 newAllowance =
+                sub96(
+                    spenderAllowance,
+                    amount,
+                    "RadicleToken::burnFrom: burned amount exceeds spender allowance"
```
Comment (cloudhead) [inline contracts/Governance/RadicleToken.sol:201]:
```suggestion
                    "RadicleToken::burnFrom: burn amount exceeds allowance"
```

File: contracts/Governance/RadicleToken.sol [hunk 3]
```diff
@@ -187,6 +187,35 @@ contract RadicleToken {
...
+        }
+
+        balances[account] = sub96(
+            balances[account],
+            amount,
+            "RadicleToken::_burnTokens: burned amount exceeds balance"
+        );
+        emit Transfer(account, address(0), amount);
+
+        _moveDelegates(delegates[account], address(0), amount);
```
Comment (cloudhead) [inline contracts/Governance/RadicleToken.sol:214]: I'm trying to wrap my head around this. Does this mean the user delegates to the zero address?

Comment (cloudhead) [reply contracts/Governance/RadicleToken.sol:214]: Looking at `_moveDelegates`, I think this is correct :thinking:

Comment (CodeSandwich) [reply contracts/Governance/RadicleToken.sol:214]: Conceptually yes, the voting power is burned along the tokens. In practice `_moveDelegates` checks for zero and doesn't apply anything to the zero address, only removes from the burner.

File: contracts/Governance/RadicleToken.sol (hunk 1 above)
Comment (CodeSandwich) [reply contracts/Governance/RadicleToken.sol]: Fixed

File: contracts/Governance/RadicleToken.sol (hunk 2 above)
Comment (CodeSandwich) [reply contracts/Governance/RadicleToken.sol]: Fixed

**Review by cloudhead — APPROVED** [review]
Nice.

File: contracts/Governance/RadicleToken.sol [hunk 4]
```diff
@@ -187,6 +187,35 @@ contract RadicleToken {
         return true;
     }
 
+    function burnFrom(address account, uint256 rawAmount) public virtual {
+        require(account != address(0), "RadicleToken::burnFrom: cannot burn from the zero address");
+        uint96 amount = safe96(rawAmount, "RadicleToken::approve: amount exceeds 96 bits");
```
Comment (cloudhead) [inline contracts/Governance/RadicleToken.sol:192]:
```suggestion
        uint96 amount = safe96(rawAmount, "RadicleToken::burnFrom: amount exceeds 96 bits");
```

Comment (cloudhead) [reply contracts/Governance/RadicleToken.sol:192]: Just noticed this one

Comment (CodeSandwich) [reply contracts/Governance/RadicleToken.sol]: Fixed. I've double checked all the new error messages, that was the last issue.

**Pull request merged**